### PR TITLE
feat(oauth): limited discord server sign-in

### LINKED ISSUE #338: 🚀 Feature: Limiting Discord login to members of a specific server
### 🔖 Feature description

Currently, from my understanding, when using Discord login, anyone with a Discord account can login/register and use Pingvin-Share. I'd like to have an option to allow only users from a specific server (guild) to register.

### 🎤 Pitch

Limiting Discord login only to XYZ server would be a great way to share files with people from the server you want. It would be easier for user management, and the members wouldn't need to have a separate login (people would be more inclined to join and share). Social login for Nextcloud is an example of a plugin that supports it.

### PR DESCRIPTION
**Pull request by zz5840**
close #338

### CODE REVIEW

File: backend/src/oauth/exceptions/errorPage.exception.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ export class ErrorPageException extends Error {
    */
   constructor(
     public readonly key: string = "default",
-    public readonly redirect: string = "/",
+    public readonly redirect: string = undefined,
```
Comment (stonith404) [inline backend/src/oauth/exceptions/errorPage.exception.ts:10]:
```suggestion
    public readonly redirect? : string,
```

Comment (zz5840) [reply backend/src/oauth/exceptions/errorPage.exception.ts]: Ok, fixed

**Review by stonith404 — APPROVED** [review]
The code looks good to me. I can't really test it as I don't have an own Discord server but if you've tested it properly we can merge it.

**Comment by zz5840** [comment]
@stonith404 Can you build a dev image please? I will test it on my server tomorrow.

**Comment by stonith404** [comment]
@zz5840 Done 👍

**Comment by zz5840** [comment]
@stonith404 I've tested it on my server, and it works as expected. Hope we can merge it.

**Comment by stonith404** [comment]
Sorry for the late response. Should I wait for your second PR before creating a new release?

**Comment by stonith404** [comment]
@zz5840 Great, I've created the release. Thank you for the contribution :)

**Pull request merged**